### ISSUE TITLE
--with-sqlcipher option on Windows x64 UCRT 

### ISSUE THREAD

**Issue by ls5302** [sparklemotion/sqlite3-ruby#516]
I'm currently porting my code to 64-bit, with 32-bit Ruby on windows I am able to run:

```
gem install -V sqlite3 -- --with-sqlcipher
```

and see `Building native extensions with: '--with-sqlcipher'`

Using Ruby 64-bit (ucrt) I don't see that message and the database I create is not encrypted.

For x86 environment the package was installed using `pacman --noconfirm -S mingw-w64-i686-sqlcipher`
For x64 environment the package was installed using `pacman --noconfirm -S mingw-w64-ucrt-x86_64-sqlcipher`

What am I doing wrong? Please help.

**Comment by flavorjones** [comment]
@ls5302 Sorry you're having problems. I'll try to help.

### first try this

My best guess without more information: you're installing the precompiled version of the gem (which is not available for 32-bit systems, but is the default for 64-bit systems).

We've got instructions on how to avoid the precompiled gem at https://github.com/sparklemotion/sqlite3-ruby/blob/main/INSTALLATION.md#avoiding-the-precompiled-native-gem

TLDR, to make sure you compile against sqlcipher during gem installation, you need to specify `--platform=ruby`

So try:

```
gem install sqlite3 --platform=ruby -- --with-sqlcipher
```

Note that the `--platform` flag must be _before_ the `--`, and the `sqlcipher` flag must be _after_ the `--`.

### else do this

If that doesn't work, I'll need more information in order to diagnose what's going on, so please reply with:
- the output of `gem env`
- the command you're running, and the full output of the command

**Comment by ls5302** [comment]
Thank you for your reply.

`gem install sqlite3 --platform=ruby -- --with-sqlcipher` works, as does adding `force_ruby_platform: true` to the Gemfile which is what I've gone with.

Having double-checked the 32-bit version is not available which explains why I see different behaviour, and now need the extra flag for the 64-bit version which is available in the GEM repo.

Many thanks!
